app/forms/otp_delivery_selection_form.rb

```diff
@@ -0,0 +1,31 @@
+class OtpDeliverySelectionForm
+  include ActiveModel::Model
+
+  attr_reader :otp_method
+
+  validates :otp_method, inclusion: { in: %w(sms voice) }
+
+  def submit(params)
+    self.otp_method = params[:otp_method]
+    self.resend = params[:resend]
+
+    @success = valid?
+
+    result
+  end
+
+  private
+
+  attr_writer :otp_method
+  attr_accessor :resend
+  attr_reader :success
+
+  def result
+    {
+      success?: success,
+      delivery_method: otp_method,
+      resend?: resend,
```

Comment:
By passing in a `resend` params when the user clicks the "Send me a new code" link, we can capture whether or not the delivery was the result of a resend request or not, which means we only need to call `analytics.track_event` once in the controller instead of twice. This allows us to remove the duplication in the `new` and `send_code` methods that we used to have in TwoFactorAuthenticationController.